### PR TITLE
[GH-917] Character visual feedback in damage zone

### LINKED ISSUE #917: [DESIGN] Enemies should show some kind of feedback when in damage zone
## 🎨 Design
After the merge of #882 (Cel Shading + Lightning Improvements) I can see feedback for when I enter the damage zone, but if an enemy enters damage zone he seems to be floating over this. We should add some kind of feedback so that it seems he is under it

https://github.com/lambdaclass/curse_of_myrra/assets/22064054/5b129cec-7acf-4a6b-920e-36311aca88cf

### PR DESCRIPTION
**Pull request by tkz00**
Closes #917 

## Motivation

Apart from the characters model flashing repeatedly there was no visual feedback to indicate a player was receiving damage when out of bounds, when the character that is out of bounds is the one we are controlling we would get a screen filter as a visual feedback, but that wasn't enough and also it was unclear if enemies were taking damage since this canvas filter didn't apply to them. This PR aims to solve that issue by adding a red particle effect to all characters when in the damage zone.

## Summary of changes

Created a new state feedback prefab called OutOfArea and added it to the FeedbackContainer. Referenced this component in the feedbacksStatesPrefabs list of the CharacterFeedbacks script inside each of the characters prefabs.

## How has this been tested?

In a match with two players, one of the clients with this PR's branch in my pc and the other in my phone with the testing branch, moved both the phone and pc characters into the damage zone and checked the desired particle effects where displayed in my pc's client.

## Test additions / changes

No changes or additions were made to the tests in this PR.

## Checklist
- [X] I have tested the changes locally.
- [X] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [ ] Tested in Android.

### CODE REVIEW

**Review by BertovDev — CHANGES_REQUESTED** [review]
Nice effect! and only one thing to correct.

1. When player dies in the zone the feedback is still playing. This can be easy fixed doing the following:
In the `SetPlayerDead` method in `Battle.cs` add the following code:
```
 playerCharacter
            .GetComponent<CharacterFeedbacks>()
            .ClearAllFeedbacks(playerCharacter.gameObject);
```

**Comment by tkz00** [comment]
> Nice effect! and only one thing to correct.
> 
>     1. When player dies in the zone the feedback is still playing. This can be easy fixed doing the following:
>        In the `SetPlayerDead` method in `Battle.cs` add the following code:
> 
> 
> ```
>  playerCharacter
>             .GetComponent<CharacterFeedbacks>()
>             .ClearAllFeedbacks(playerCharacter.gameObject);
> ```

You are right, already made the changes!

**Comment by BertovDev** [comment]
> > Nice effect! and only one thing to correct.
> > ```
> > 1. When player dies in the zone the feedback is still playing. This can be easy fixed doing the following:
> >    In the `SetPlayerDead` method in `Battle.cs` add the following code:
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> >  playerCharacter
> >             .GetComponent<CharacterFeedbacks>()
> >             .ClearAllFeedbacks(playerCharacter.gameObject);
> > ```
> 
> You are right, already made the changes!

Cool and you also avoid multiple GetComponent<> calls with this add. Nice!!

**Review by BertovDev — APPROVED** [review]

File: client/mono_crash.mem.92026.1.blob
Comment (BertovDev) [inline client/mono_crash.mem.92026.1.blob:1]: Delte this

**Review by BertovDev — CHANGES_REQUESTED** [review]
Monocrash😞

**Review by BertovDev — DISMISSED** [review]

**Review by BertovDev — APPROVED** [review]
LGTM!

**Review by ncontinanza — APPROVED** [review]

**Pull request merged**